File: 2.0/java/contacts/createContactList.java
import com.callfire.api.client.CallfireClient;
import com.callfire.api.client.api.common.model.ResourceId;
import com.callfire.api.client.api.contacts.model.Contact;
import com.callfire.api.client.api.contacts.model.request.CreateContactListRequest;

import java.util.Arrays;

class ApiClientSample {
    public static void main(String[] args) {
        CallfireClient client = new CallfireClient("api login", "api password");
        CreateContactListRequest request1 = CreateContactListRequest.<String>create()
            .name("My Contact List")
            .contacts(Arrays.asList("12132212384", "12136612355", "12133312300"))
            .contactNumbersField("workPhone")
            .build();
        ResourceId resourceId1 = client.contactListsApi().create(request1);

        // second example shows how to create contact list from existing contacts, what we need is just their ids:
        CreateContactListRequest request2 = CreateContactListRequest.<Long>create()
            .name("My Contact List")
            .contacts(Arrays.asList(800634185003L, 800734186003L, 800834187003L, 800984185003L))
            .build();
        ResourceId resourceId2 = client.contactListsApi().create(request2);

        // next code sample uses Contact object in CreateContactListRequest request:
        Contact contact1 = new Contact();
        contact1.setFirstName("Alice");
        contact1.setLastName("Moore");
        contact1.setHomePhone("12135551126");

        Contact contact2 = new Contact();
        contact1.setFirstName("Bob");
        contact1.setLastName("Smith");
        contact1.setHomePhone("12135551127");
        contact1.getProperties().put("age", "30");
        CreateContactListRequest request3 = CreateContactListRequest.<Contact>create()
            .name("My Contact List")
            .contacts(Arrays.asList(contact1, contact2))
            .build();
        ResourceId resourceId3 = client.contactListsApi().create(request3);
    }
}

File: 2.0/csharp/contacts/createContactList.cs
using System.Collections.Generic;
using CallfireApiClient;
using CallfireApiClient.Api.Common.Model;
using CallfireApiClient.Api.Contacts.Model;
using CallfireApiClient.Api.Contacts.Model.Request;

public class ApiClientSample
{
    public static void Main(string[] args)
    {
        var client = new CallfireClient("api_login", "api_password");

        var request1 = new CreateContactListRequest<Contact>
        {
            Contacts = new List<Contact>
            {
                new Contact
                {
                    FirstName = "Alice",
                    LastName = "Moore",
                    HomePhone = "12135551126"
                },
                new Contact
                {
                    FirstName = "Bob",
                    LastName = "Smith",
                    HomePhone = "12135551127",
                    Properties = new Dictionary<string, string>
                    {
                        {"age", "30"}
                    }
                }
            },
            Name = "My Contact List"
        };
        ResourceId resourceId1 = client.ContactListsApi.Create(request1);

        var request2 = new CreateContactListRequest<string>
        {
            Contacts = new List<string> {"12132212384", "12136612355", "12133312300"},
            ContactNumbersField = "workPhone",
            Name = "My Contact List"
        };
        ResourceId resourceId2 = client.ContactListsApi.Create(request2);

        var request3 = new CreateContactListRequest<long>
        {
            Contacts = new List<long> {800634185003, 800734186003, 800834187003, 800984185003},
            Name = "My Contact List"
        };
        ResourceId resourceId3 = client.ContactListsApi.Create(request3);
    }
}
